<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pycqa/pydocstyle.git.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://gitlab.com/PyCQA/flake8.
[INFO] Initializing environment for https://gitlab.com/PyCQA/flake8:pep8-naming.
[INFO] Initializing environment for https://github.com/mgedmin/check-manifest.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pycqa/pydocstyle.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://gitlab.com/PyCQA/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/mgedmin/check-manifest.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Failed
- hook id: remove-tabs
- exit code: 1
- files were modified by this hook

Substituting tabs in: data/commits.train by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit
Substituting tabs in: data/commits-labeled.txt by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit
Substituting tabs in: .github/CODEOWNERS by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit
Substituting tabs in: output/output.csv by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

Trim Trailing Whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing data/commits-labeled.txt
Fixing data/commits.train
Fixing thoth/glyph/cli.py
Fixing OWNERS

Check for merge conflicts................................................Passed
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing .github/PULL_REQUEST_TEMPLATE.md
Fixing mypy.ini

Tests should end in _test.py.............................................Failed
- hook id: name-tests-test
- exit code: 1

tests/test_cli.py does not match pattern ".*_test.py"

Check for added large files..............................................Passed
Check for byte-order marker..............................................Passed
Check for case conflicts.................................................Passed
Check docstring is first.................................................Passed
Check JSON...............................................................Passed
Check for broken symlinks................................................Passed
Detect Private Key.......................................................Passed
Check python ast.........................................................Passed
Debug Statements (Python)................................................Passed
pydocstyle...............................................................Failed
- hook id: pydocstyle
- exit code: 1

setup.py:1 at module level:
        D100: Missing docstring in public module
setup.py:8 in public function `get_install_requires`:
        D103: Missing docstring in public function
setup.py:14 in public function `get_version`:
        D103: Missing docstring in public function
setup.py:25 in public function `read`:
        D103: Missing docstring in public function
setup.py:45 in public method `initialize_options`:
        D102: Missing docstring in public method
setup.py:49 in public method `finalize_options`:
        D102: Missing docstring in public method
setup.py:54 in public method `run_tests`:
        D102: Missing docstring in public method

Check Toml...............................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

thoth/glyph/cli.py:35:1: error: Function is missing a return type annotation  [no-untyped-def]
thoth/glyph/cli.py:35:1: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
thoth/glyph/cli.py:48:1: error: Function is missing a return type annotation  [no-untyped-def]
Found 3 errors in 1 file (checked 2 source files)

black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted /workspace/repo/thoth/glyph/cli.py
reformatted /workspace/repo/setup.py
All done! ✨ 🍰 ✨
2 files reformatted, 2 files left unchanged.

flake8...................................................................Failed
- hook id: flake8
- exit code: 1

setup.py:3:1: F401 'setuptools.find_packages' imported but unused
setup.py:5:2: N812 lowercase 'test' imported as non lowercase 'TestCommand'

check-manifest...........................................................Failed
- hook id: check-manifest
- exit code: 1

lists of files in version control and sdist do not match!
missing from sdist:
  .pre-commit-config.yaml
  .thoth.yaml
  CHANGELOG.md
  code
  code/commit-classifier.ipynb
  data
  data/commits-labeled.txt
  data/commits.train
  data/model_commits_v2_quant.bin
  output
  output/output.csv
  tests
  tests/test_cli.py
  thoth-glyph
suggested MANIFEST.in rules:
  include *.md
  include *.yaml
  recursive-include code *.ipynb
  recursive-include data *.bin
  recursive-include data *.train
  recursive-include data *.txt
  recursive-include output *.csv
  recursive-include tests *.py
```
</details>

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,4 @@
 click
 thoth-common
+pandas
+pygit2

--- a/thoth/glyph/cli.py
+++ b/thoth/glyph/cli.py
@@ -37,7 +37,7 @@ init_logging()
 
 _LOGGER = logging.getLogger("glyph")
 
-MODEL_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) + "/data/model_commits_v2_quant.bin"
+DEFAULT_MODEL_PATH = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))), "data/model_commits_v2_quant.bin")
 
 def _print_version(ctx: click.Context, _, value: str):
     """Print glyph version and exit."""
@@ -108,8 +108,8 @@ def generate(output: str) -> None:
 def classify(message: str) -> None:
     """Generate CHANGELOG entries from the current Git project."""
     _LOGGER.info("Hello, glyph!")
-    _LOGGER.info("Model Path : " + MODEL_PATH)
-    classifier = load_model(MODEL_PATH) 
+    _LOGGER.info("Model Path : " + DEFAULT_MODEL_PATH)
+    classifier = load_model(DEFAULT_MODEL_PATH) 
     label = classifier.predict(message.lower())
     click.echo("Label : " + str(label[0][0])[8:])
 
@@ -124,24 +124,35 @@ def classify(message: str) -> None:
 @click.option(
     "--start",
     type=str,
-    default="",
     help="Starting date"
 )
 @click.option(
     "--end",
     type=str,
-    default="",
     help="End date"
 )
-def classifyrepo(path: str, start: str, end: str) -> None:
-    _LOGGER.info("Model Path : " + MODEL_PATH)
-    classifier = load_model(MODEL_PATH)
-
+@click.option(
+    "--output",
+    type=str,
+    help="Generated output file"
+)
+@click.option(
+    "--model",
+    type=str,
+    help="Generated output file"
+)
+def classifyrepo(path: str, start: str, end: str, output: str, model:str) -> None:
+    if model is None:
+        model = DEFAULT_MODEL_PATH
+    _LOGGER.info("Model Path : " + model)
+    classifier = load_model(model)
     start_time = 0
     end_time = 4119206400
 
-    if(start != "" and end != ""):
+    if start is not None:
         start_time = int(time.mktime(datetime.datetime.strptime(start, "%Y-%m-%d").timetuple()))
+
+    if end is not None:
         end_time = int(time.mktime(datetime.datetime.strptime(end, "%Y-%m-%d").timetuple()))
 
     repo = Repository(path + "/.git")
@@ -159,8 +170,10 @@ def classifyrepo(path: str, start: str, end: str) -> None:
     res_list = [x[0] for x in res]
     lst2 = [item[0] for item in res_list]
     df['labels_predicted'] = lst2
-
-    df.to_csv("output.csv", sep='\t')
+    if output is None:
+        df.to_csv("output.csv", sep='\t')
+    else:
+        df.to_csv(output, sep='\t')
 
 
 

--- a/thoth/glyph/cli.py
+++ b/thoth/glyph/cli.py
@@ -26,6 +26,13 @@ from thoth.common import init_logging
 from thoth.glyph import __version__ as glyph_version
 from fasttext import load_model
 
+import pandas as pd
+from pygit2 import Repository
+from pygit2 import GIT_SORT_TOPOLOGICAL, GIT_SORT_REVERSE
+
+import time
+import datetime
+
 init_logging()
 
 _LOGGER = logging.getLogger("glyph")
@@ -105,6 +112,56 @@ def classify(message: str) -> None:
     classifier = load_model(MODEL_PATH) 
     label = classifier.predict(message.lower())
     click.echo("Label : " + str(label[0][0])[8:])
+
+@cli.command()
+@click.option(
+    "--path",
+    "-p",
+    type=str,
+    required=True,
+    help="Path to Git repository"
+)
+@click.option(
+    "--start",
+    type=str,
+    default="",
+    help="Starting date"
+)
+@click.option(
+    "--end",
+    type=str,
+    default="",
+    help="End date"
+)
+def classifyrepo(path: str, start: str, end: str) -> None:
+    _LOGGER.info("Model Path : " + MODEL_PATH)
+    classifier = load_model(MODEL_PATH)
+
+    start_time = 0
+    end_time = 4119206400
+
+    if(start != "" and end != ""):
+        start_time = int(time.mktime(datetime.datetime.strptime(start, "%Y-%m-%d").timetuple()))
+        end_time = int(time.mktime(datetime.datetime.strptime(end, "%Y-%m-%d").timetuple()))
+
+    repo = Repository(path + "/.git")
+    orig_messages = []
+    for commit in repo.walk(repo.head.target, GIT_SORT_TOPOLOGICAL):
+        if(commit.commit_time > start_time and commit.commit_time < end_time):
+             orig_messages.append(commit.message.lower())
+
+    print(str(len(orig_messages)) + " commits classified")
+    df = pd.DataFrame(orig_messages, columns = ['message'])
+    df = df.replace('\n','', regex=True)
+    commits = list(df['message'].astype(str))
+    labels = classifier.predict(commits)
+    res = list(zip(*labels))
+    res_list = [x[0] for x in res]
+    lst2 = [item[0] for item in res_list]
+    df['labels_predicted'] = lst2
+
+    df.to_csv("output.csv", sep='\t')
+
 
 
 __name__ == "__main__" and cli()

